### PR TITLE
fix race condition with loading events

### PR DESCRIPTION
**Pull request by patricioe**
``` INFO [split-segmentFetcher-0] (SDKReadinessGates.java:110) - Registered segment: demo_orgs
 INFO [split-splitFetcher-0] (SDKReadinessGates.java:62) - splits are ready
 INFO [split-segmentFetcher-0] (RefreshableSegment.java:135) - demo_orgs added keys: [2dfcc270-aacc-11e8-a8d2-xxxxxxxxxxx, 7479b040-aa14-11e8-91c4-xxxxxxxxxxx, 745622b0-aa14-11e8-91c4-12bc02531368... 5 others]
 INFO [split-segmentFetcher-0] (SDKReadinessGates.java:130) - demo_orgs segment is ready```

### CODE REVIEW

File: client/pom.xml [hunk 1]
```diff
@@ -156,5 +156,11 @@
             <version>1.10.19</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
```
Comment (SenhorCastor) [inline client/pom.xml]: Is this intended?

File: client/src/test/resources/log4j.properties [hunk 2]
```diff
@@ -0,0 +1,15 @@
+log4j.rootLogger=info, stdout
```
Comment (SenhorCastor) [inline client/src/test/resources/log4j.properties]: is this file intended\?

File: client/pom.xml (hunk 1 above)
Comment (patricioe) [reply client/pom.xml]: yes, for testing. Otherwise there are no logs in the console.

File: client/src/test/resources/log4j.properties (hunk 2 above)
Comment (patricioe) [reply client/src/test/resources/log4j.properties]: yes
